lms/djangoapps/courseware/access.py

```diff
@@ -314,10 +340,10 @@ def see_exists():
             # seen by non-staff
             if course.ispublic:
                 debug("Allow: ACCESS_REQUIRE_STAFF_FOR_COURSE and ispublic")
-                return True
+                return ACCESS_GRANTED
             return _has_staff_access_to_descriptor(user, course, course.id)
 
-        return can_enroll() or can_load()
+        return ACCESS_GRANTED if can_enroll() or can_load() else ACCESS_DENIED
```

Comment:
Nit: consider adding a parenthesis around `can_enroll() or can_load()` for readability:

```
return ACCESS_GRANTED if (can_enroll() or can_load()) else ACCESS_DENIED
```